axiom/nr_errors.c

```diff
@@ -97,6 +170,25 @@ void nr_error_destroy(nr_error_t** error_ptr) {
   nr_realfree((void**)error_ptr);
 }
 
+void nr_error_destroy_additional_params(nr_error_t** error_ptr) {
+  nr_error_t* error;
+
+  if (0 == error_ptr) {
+    return;
+  }
+  error = *error_ptr;
+  if (0 == error) {
+    return;
+  }
+  nr_free(error->message);
+  nr_free(error->klass);
+  nr_free(error->error_file);
+  nr_free(error->error_context);
+  nr_free(error->span_id);
+  nr_free(error->stacktrace_json);
+  nr_realfree((void**)error_ptr);
```

Comment:
Based on the name of this function, I would expect it to only destroy the additional params, and thus still require the calling of `nr_error_destroy` afterward to fully free the error.

Reply:
To reduce code duplication, I have removed this function and added the additional logic in `nr_error_destroy` inside of a if statement: [f917ced](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/f917ced797ecc2b0f2f560487081e73bf4cf47a1)